lighthouse-core/lib/page-functions.js

```diff
@@ -170,14 +170,14 @@ function getOuterHTMLSnippet(element, ignoreAttrs = [], snippetCharacterLimit =
 /**
  * Get the maximum size of a texture the GPU can handle
  * @see https://bugs.chromium.org/p/chromium/issues/detail?id=770769#c13
+ * @return {number}
  */
-/* istanbul ignore next */
 function getMaxTextureSize() {
   try {
-    let canvas = document.createElement('canvas');
-    let gl = canvas.getContext('webgl');
+    const canvas = document.createElement('canvas');
+    const gl = canvas.getContext('webgl');
+    if (!gl) throw new Error('no webgl');
     const maxTextureSize = gl.getParameter(gl.MAX_TEXTURE_SIZE);
-    canvas = gl = undefined; // Cleanup for GC
```

Comment:
I didn't understand why this was necessary at the time because I thought escape analysis is the same either way and the memory leak potential is in the side-effects of creating a webgl canvas creating a reference (in which case `canvas = gl = undefined` does nothing to help that, right?), but supposedly there was still a reason for this.

Given how fresh it is though, maybe @paulirish can clarify where https://github.com/GoogleChrome/lighthouse/pull/11847#discussion_r556177616 ended up from his perspective :)

Reply:
I think it started from this offhand comment, not a detected problem https://github.com/GoogleChrome/lighthouse/pull/11847#discussion_r544904838 

But yeah, they're references to DOM objects. Unsetting the locals isn't going to affect their lifecycle.